app/controllers/pages_controller.rb

```diff
@@ -1,2 +1,17 @@
 class PagesController < ApplicationController
+  include PageableFeedConcern
+
+  before_action :authenticate_user!
+
+  def home
+    respond_to do |format|
+      format.html { initialize_pageable_feed }
+      format.js {
+        @feed_item = FeedItemPresenter.new(
+          feed_item: current_user.toggle_feed_item_recommended(params[:id]),
+          current_user: current_user,
+        )
+      }
+    end
+  end
```

Comment:
Simple, maintainable. Postbacks result from page navigation - pull only those feeds we need to appear in the UI, NOT the whole database with every pull. Ajax calls simply need to retrieve the feed item that is being recommended (starred); unobtrusive javascript allows us to only update that SINGLE html element in the UI, making for a much better user experience.